Streamline counter, wCCD and voting smart contract tutorials (#1043)
Concordium/concordium.github.io: source/mainnet/smart-contracts/tutorials/voting/voting-dapp.rst

.. _voting-frontend:

========================
Setting up the frontend
========================

This is the second part of :ref:`the voting smart contract tutorial<voting-sc>`. In this part of the tutorial,
we will create a web frontend application to make it easier for users to interact with the smart contract.

Non-tech users will likely not be able to interact with smart contracts via a Concordium node
and some users may choose not to host their own Concordium node locally. This tutorial part shows you
a setup that eliminates the need for the user to host their own Concordium node.

You can lower the bar for entry by coding an appealing frontend that provides additional information
to your users. Web frontends are a familiar sight nowadays, but to use
the frontend, users will also need to download the |bw| (or simply "the browser wallet") as a browser extension.
The installation and safe usage of the browser wallet might be new for people using your frontend.

Providing comprehensive explanations and step-by-step guides on your website on topics

> related to the browser wallet is important for a good user experience. The browser wallet

Reviewer: I agree with you that the naming `browser wallet` is much more natural than `|bw| = Wallet for Web` and I did not support the naming decision in the past. We need to get a consensus in the company to rename it throughout the documentation since in the past people agreed and pushed for calling it `Wallet for Web`.

As such, we should change the |bw| to `browser wallet` (if agreed internally) instead of partially using the constants and plain `browser wallet` in the text. The current mix will make it hard to have consistent naming throughout the documentation.

Author: My biggest complaint is that `|bw|` expands to `Concordium Wallet for Web` which is just _super long_ and really only makes sense when referring to the wallet in a title or maybe the first time you mention it. I think having `|bw|` expand to that is okay, but it makes no sense to use that expansion everywhere in the prose, as it makes the writing very unnatural.

I don't even know if using such expansions provide a big benefit, finding stuff by a simple search or regex is fine I think? But yea we should agree on how we refer to these things.

Author: To be clear, I am not advocating for changing the official name of the wallet or anything. I am using "browser wallet" like a pronoun or a shorthand to refer to "Concordium Wallet for Web".

I personally find that "web wallet" is a bit of an inaccurate name though, as it is not a web application, but a browser extension.

|bw|
-------------------------

> Concordium's browser wallet is an extension to Chromium-based browsers such as Google Chrome.

Reviewer: everywhere in the documentation:

```suggestion
Concordium's |bw| is an extension to Chromium-based browsers such as Google Chrome.
```

Author: This will expand to "Concordium's Concordium Wallet for Web", but even fixing the double Concordium, I think the long title just sounds super unnatural in a casual sentence like this.

The browser wallet allows you to interact with the chain and make transactions, by connecting to a (gRPC) server that communicates with a node.
The browser wallet hosts the private keys corresponding to the accounts of the user.

Your frontend code that is run in the browser constructs the transaction object
and sends it to the browser wallet extension. The transaction object is signed by the private key hosted in the wallet
and transmitted to the server via HTTPS. This server has access to a Concordium node and converts
the request (including the signed transaction object) that comes via HTTPS from the browser wallet
to a request that the Concordium node can execute. The signed transaction is
transmitted via peer-to-peer communication to other Concordium nodes and becomes
part of the Concordium blockchain.

Installation

> -------------------------

Reviewer: Should have the same length as the heading
```suggestion
--------------
```

    .. image:: ../wCCD/images/wCCD_tutorial_13.png
        :width: 30 %

Create a new account on testnet by going through the setup steps.
You have to choose a password to secure the wallet.
This password is needed to log in to the wallet.

.. image:: ../wCCD/images/wCCD_tutorial_15.png
    :width: 30 %

The wallet creates a unique secret recovery phrase. Write down the secret recovery phrase
and keep it in a safe place to be able to recover your accounts in case
you lose access to your device.

You have now completed the setup. Please verify that your wallet is connected to the Concordium Testnet.

.. image:: ../wCCD/images/wCCD_tutorial_16.png
    :width: 30 %

.. image:: ../wCCD/images/wCCD_tutorial_17.png
    :width: 30 %

Before you can create a new account, you need to create an identity. You can do this easily on testnet via the Concordium testnet ID provider.

.. image:: ../wCCD/images/wCCD_tutorial_19.png
    :width: 30 %

.. image:: ../wCCD/images/wCCD_tutorial_20.png
    :width: 30 %

.. image:: ../wCCD/images/wCCD_tutorial_21.png
    :width: 30 %

.. image:: ../wCCD/images/wCCD_tutorial_22.png
    :width: 30 %

> Finally, make a new account on testnet.

Reviewer: ```suggestion
Finally, create a new account on testnet.
```

Author: Quite the nitpick but fine :sweat_smile:

.. image:: ../wCCD/images/wCCD_tutorial_19.png
    :width: 30 %

.. image:: ../wCCD/images/wCCD_tutorial_20.png
    :width: 30 %

> For testing you'll need some CCD in your account. Send some CCD to your new account or request some CCD from the :ref:`testnet faucet button<testnet-faucet>`.

Reviewer: Testnet faucets are usually websites, but our faucet (where the link points to) is built in the wallet itself (this is rather unusual) and as such good to mention since people expect a website to pop up.

```suggestion
For testing you'll need some CCD in your account. Send some CCD to your new account or request some CCD from the :ref:`testnet faucet button<testnet-faucet>` within the |bw|.
```
